### PR TITLE
Send records to ruby in chunks

### PR DESCRIPTION
**Pull request by mikowitz**
* Send records to Ruby in chunks to improve timing
* Refactors `stdout`/`stderr` logging to not drop log statements.

### CODE REVIEW

File: lambda/index.js [hunk 1]
```diff
@@ -1,40 +1,39 @@
...
-      console.log(x);
-    });
-  });
-  child.stderr.on('data', function (data) {
-    data.split("\n").forEach(function(x) {
-      console.log(x);
-    });
-  });
-};
+const exec = require('child_process').execFile;
```
Comment (jaeger401) [inline lambda/index.js]: Sorry, this is my typo from the code snippet I sent -- `exec` should probably be `execFile` for clarity. The call below would have to change as well.

**Review by jaeger401 — APPROVED** [review]
Just one thing about naming that's my fault, otherwise LGTM.

**Comment by mikowitz** [comment]
@jaeger401 does that last change match what you were doing in your testing?

File: lambda/index.js [hunk 2]
```diff
@@ -1,40 +1,39 @@
...
-      console.log(x);
-    });
-  });
-};
+const execFile = require('child_process').execFile;
 
 exports.handler = function(event, context) {
+  const chunkSize = 5;
+
+  var records = event['Records'];
```
Comment (bjackson2) [inline lambda/index.js]: Can this be `event.Records`?

File: lambda/index.js [hunk 3]
```diff
@@ -1,40 +1,39 @@
...
-      cb();
-    });
-  });
+  console.log("Received " + records.length + " records.")
 
-  q.on('success', function(result, job) {
-    console.log('job finished processing:', job.toString().replace(/\n/g, ''));
-  });
+  var chunkNo = 0;
+  var chunks = []
```
Comment (bjackson2) [inline lambda/index.js]: rzl3afeevnnz9enpqs6w

File: lambda/index.js [hunk 4]
```diff
@@ -1,40 +1,39 @@
...
+
+  while (chunkNo < records.length) {
```
Comment (bjackson2) [inline lambda/index.js]: This is kinda like an index/limit/offset - thoughts on using naming like that?

File: lambda/index.js [hunk 5]
```diff
@@ -1,40 +1,39 @@
...
-    console.log('all done!');
+  chunks.forEach(function(chunk, index) {
+      const child = execFile('./ruby_wrapper', [JSON.stringify(chunk), JSON.stringify(context)], {}, (error, stdout, stderr) => {
+      if (error) {
+        console.error(`exec error: ${error}`);
+        return;
+      }
+      stdout.trim().split("\n").forEach(function(x) {
+        log = x.trim();
+        if (log != "") {
```
Comment (bjackson2) [inline lambda/index.js]: I'd use a `!==` if possible here

File: lambda/index.js [hunk 6]
```diff
@@ -1,40 +1,39 @@
...
+          console.log(log);
+        }
+      });
+      stderr.trim().split("\n").forEach(function(x) {
+        log = x.trim();
+        if (log != "") {
```
Comment (bjackson2) [inline lambda/index.js]: ditto

**Review by bjackson2 — APPROVED** [review]

**Pull request merged**